### DISCUSSION TITLE
Is it possible to lock files when EngineCore is up?

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#77] [Ideas]
Seems like we could start different instance simultaneously and change the files or cause phantom read... etc.

**Comment by ingted** [comment]
Reproduce:

Start two instance with different port, modify data in one instance and then restart another instance, now the change could be aware in another instance.

**Comment by NTDLS** [comment]
Yes, it would not be wise to modify the files outside of the engine as the engine could have the file cached, which could lead to one instance overwriting data from another instance or not knowing about "fresh" data as it is relying on the cache. Unfortunately, locking all the files would be a very resource intensive process. The StackOverflow database I have resulted in hundreds of thousands of files which would be a lot of handles for locking files.

We could however look into storing the files in a VHD, programmatically mounting that VHD and locking the volume with a single call to CreateFile/FILE_SHARE_NONE. I believe this approach is supported on Linux as well.

That said, I think there are just some things users should be aware not to do.

**Comment by ingted** [comment]
```
That said, I think there are just some things users should be aware not to do.
```
